### PR TITLE
[Schedules] Change default labels value to None and add enrichment with db labels

### PR DESCRIPTION
**Pull request by rokatyy**
This PR addresses an issue where `schedule.labels` were cleaned up when updating a schedule with no labels passed. Additionally, I have updated the `test_update_schedule` test by adding patching functionality to enforce label enrichment. Without this fix, the test fails.

Jira - https://iguazio.atlassian.net/browse/ML-7019

### CODE REVIEW

File: tests/api/utils/test_scheduler.py [hunk 1]
```diff
@@ -1089,13 +1089,16 @@ async def test_schedule_convert_from_old_credentials_to_new(
     _assert_schedule_secrets(scheduler, project, schedule_name, None, None)
 
 
+@unittest.mock.patch.object(Scheduler, "_store_schedule_secrets_using_auth_secret")
```
Comment (alonmr) [inline tests/api/utils/test_scheduler.py]: I think you can add the return value here no?
```suggestion
@unittest.mock.patch.object(Scheduler, "_store_schedule_secrets_using_auth_secret", return_value="auth-secret")
```

**Review by alonmr — COMMENTED** [review]
It's important we add a unit test here. We should verify what happens when providing:
1. labels is None
2. labels is {} 
3. labels is not empty
All when we have labels in the DB of course

**Comment by rokatyy** [comment]
@alonmr that sounds exactly like what `test_update_schedule` does :)

File: tests/api/utils/test_scheduler.py (hunk 1 above)
Comment (rokatyy) [reply tests/api/utils/test_scheduler.py]: @alonmr TIL 👍

**Review by liranbg — APPROVED** [review]

**Review by alonmr — APPROVED** [review]

**Pull request merged**